application/helpers/frontend_helper.php

```diff
@@ -1897,20 +1897,30 @@ function doAssessment($surveyid, $returndataonly=false)
                 }
 
                 // If this is a question (and not a survey field, like ID), save asessment value
-                if (!empty($field['title']))
+                if ($field['qid'] > 0)
                 {
                     /**
                      * Allow Plugin to update assessment value
                      */
                     // Prepare Event Info
                     $event = new PluginEvent('afterSurveyQuestionAssessment');
                     $event->set('surveyId', $surveyid);
-                    $event->set('question', $field);
+                    $event->set('lang', $_SESSION['survey_'.$surveyid]['s_lang']);
+                    $event->set('gid', $field);
+                    $event->set('qid', $field['gid']);
+
+                    if (array_key_exists('sqid', $field))
+                    {
+                        $event->set('sqid', $field['sqid']);
+                    }
+
```

Comment:
> Adding fieldname would require parsing that all time (memory + cpu + not that comfortable). :)

I meant it would require parsig all time from the plugin side.

> The X are in aid (maybe ????)
Ohh I see what you mean. I will also send it.